src/libmongoc/src/mongoc/mongoc-ts-pool-private.h

```diff
@@ -0,0 +1,97 @@
+#ifndef MONGOC_TS_POOL_PRIVATE_H
+#define MONGOC_TS_POOL_PRIVATE_H
+
+#include <stddef.h>
+#include <stdbool.h>
```

Comment:
I'm going to remove the bool parameters, so that's just going to disappear, but the comment reminds me that I would like to discuss the project's `#include`-hygiene, not specifically related to this PR, but in general.